src/CSharp.Workspaces/CSharp/SyntaxLogicalInverter.cs

```diff
@@ -128,13 +128,16 @@ internal static SyntaxLogicalInverter GetInstance(Document document)
             case SyntaxKind.IsExpression:
                 {
                     var isExpression = (BinaryExpressionSyntax)expression;
-
-                    return IsPatternExpression(
-                        isExpression.Left,
-                        isExpression.OperatorToken.WithTrailingTrivia(Space),
-                        UnaryPattern(
-                            Token(SyntaxKind.NotKeyword).WithTrailingTrivia(isExpression.OperatorToken.TrailingTrivia),
-                            TypePattern((TypeSyntax)isExpression.Right)));
+                    var rightTypeSymbol = (ITypeSymbol)semanticModel.GetSymbolInfo(isExpression.Right, cancellationToken).Symbol!;
+                    return (Options.UseNotPattern)
+                        ? IsPatternExpression(
+                            isExpression.Left,
+                            isExpression.OperatorToken.WithTrailingTrivia(Space),
+                            UnaryPattern(
+                                Token(SyntaxKind.NotKeyword)
+                                    .WithTrailingTrivia(isExpression.OperatorToken.TrailingTrivia),
+                                TypePattern(rightTypeSymbol.ToTypeSyntax(SymbolDisplayFormat.FullyQualifiedFormat))))
```

Comment:
It's not necessary to specify format, just call `ToTypeSyntax()`

Reply:
I beleive it is when you need the FullyQualifiedFormat as this is not the default?

Reply:
The default format used in `ToTypeSyntax` uses fully qualified name (including `global` namespace).

https://github.com/JosefPihrt/Roslynator/blob/25288a7ffe53797511ee5a18203f606b046890f6/src/CSharp/CSharp/Extensions/SymbolExtensions.cs#L275

Reply:
I thought about this section one more time and if RHS expression is TypeSyntax then we can just use it as it is and we don't need to obtain symbol and then get type syntax from the symbol:

```cs
    if (Options.UseNotPattern
        && isExpression.Right is TypeSyntax type)
    {
        return IsPatternExpression(
            isExpression.Left,
            isExpression.OperatorToken.WithTrailingTrivia(Space),
            UnaryPattern(
                Token(SyntaxKind.NotKeyword)
                    .WithTrailingTrivia(isExpression.OperatorToken.TrailingTrivia),
                TypePattern(type)));
    }
    else
    {
        return DefaultInvert(expression);
    }
```

Reply:
Previous post is incorrect, please ignore it.

Following code should do the trick to apply Simplifier annotation only when possible:

Condition:
```cs
bool canUseSimplifier = SymbolEqualityComparer.Default.Equals(
    semanticModel.GetSpeculativeSymbolInfo(isExpression.SpanStart, isExpression.Right, SpeculativeBindingOption.BindAsExpression).Symbol,
    semanticModel.GetSpeculativeSymbolInfo(isExpression.SpanStart, isExpression.Right, SpeculativeBindingOption.BindAsTypeOrNamespace).Symbol);
```

Entire section:
```cs
var isExpression = (BinaryExpressionSyntax)expression;
ITypeSymbol rightTypeSymbol = semanticModel.GetTypeSymbol(isExpression.Right, cancellationToken)!;

TypeSyntax type = rightTypeSymbol.ToTypeSyntax();

if (SymbolEqualityComparer.Default.Equals(
    semanticModel.GetSpeculativeSymbolInfo(isExpression.SpanStart, isExpression.Right, SpeculativeBindingOption.BindAsExpression).Symbol,
    semanticModel.GetSpeculativeSymbolInfo(isExpression.SpanStart, isExpression.Right, SpeculativeBindingOption.BindAsTypeOrNamespace).Symbol))
{
    type = type.WithSimplifierAnnotation();
}

return (Options.UseNotPattern)
    ? IsPatternExpression(
        isExpression.Left,
        isExpression.OperatorToken.WithTrailingTrivia(Space),
        UnaryPattern(
            Token(SyntaxKind.NotKeyword)
                .WithTrailingTrivia(isExpression.OperatorToken.TrailingTrivia),
            TypePattern(type)))
    : DefaultInvert(expression);
```